autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -51,15 +59,33 @@ public void onApplicationEvent(UpdateFailedEvent event) {
         if (failed < RETRY_THRESHOLD) {
             updateFailures.put(autoscaleClusterId, failed);
             LOGGER.debug("Increased failed count '{}' for cluster '{}'", failed, cluster.getStackCrn());
+
+            if (event.getCausedBy() != null && (event.getCausedBy() instanceof ForbiddenException)) {
+                Integer forbiddencount = getForbiddenFailureCount(autoscaleClusterId) + 1;
+                UpdateFailedDetails updateFailedDetails = new UpdateFailedDetails(event.getLastExceptionTimestamp(), forbiddencount.longValue());
+                clusterService.setUpdateFailedDetails(cluster.getId(), updateFailedDetails);
+                forbiddenFailures.put(autoscaleClusterId, forbiddencount);
+                LOGGER.debug("Increased forbidden count '{}' for cluster '{}'", forbiddencount, cluster.getStackCrn());
+            }
         } else {
+            if (getForbiddenFailureCount(autoscaleClusterId) >= (RETRY_THRESHOLD - 1)) {
```

Comment:
> In the scenario that the machine user is not available - the machine user re-initialization will only be re-tried after RETRY_THRESHOLD * YarnMetricsClient.UPDATE_FAILED_INTERVAL. The machineUser to call the Yarn API itself will be tried in RETRY_THRESHOLD. Was that the intention? A much less frequent re-initialization?

Since the requirement was to provide some time for the user sync (for the newly created machineUser) to complete, I implemented it this way to provide tiem for the user sync to complete, and have a less frequent machineUser reinitialisation. 

> UpdateFailedHandler also has no information on whether the user being used is the machineUser or the endUser - both can result in a ForbiddenException.

Also, if for example, the 2nd attempt (after 1 hour) to use the machine user is successful - I don't think anything is clearing the status in the database. It probably should be cleared. The expectation is - once a machine user has been synced - it works after this. The machine user will be used after this point for each invocation - so I guess the information captured is correct in terms of the last Exception.

Ack, will add extra info to determine if a machineUser was used to query YARN or not, and also clear `UpdateFailedDetails` for the cluster after a successful YARN API call, with a machine user.